File: Baekjoon/Queue/18258.kt
import java.io.BufferedReader
import java.io.InputStreamReader
import java.util.*

fun main() {
    val br = BufferedReader(InputStreamReader(System.`in`))
    val n = br.readLine().toInt()
    val queue = LinkedList<Int>()
    val sb = StringBuffer()

    for (m in 0 until n) {
        val st = StringTokenizer(br.readLine(), " ")
        when(st.nextToken()) {
            "push" -> {
                queue.add(st.nextToken().toInt())
            }
            "pop" -> {
                if (queue.size > 0) {
                    sb.append("${queue.pollFirst()}\n")
                }
                else {
                    sb.append("-1\n")
                }
            }
            "size" -> {
                sb.append("${queue.size}\n")
            }
            "empty" -> {
                if (queue.size == 0) {
                    sb.append("1\n")
                }
                else {
                    sb.append("0\n")
                }
            }
            "front" -> {
                if (queue.size > 0) {
                    sb.append("${queue.first}\n")
                }
                else {
                    sb.append("-1\n")
                }
            }
            "back" -> {
                if (queue.size > 0) {
                    sb.append("${queue.last}\n")
                }
                else {
                    sb.append("-1\n")
                }
            }
        }
    }

    print(sb.toString())
}

File: Baekjoon/Queue/18258.py
import sys
from collections import deque

n = int(sys.stdin.readline())
queue = deque()

for _ in range(n):
    i = sys.stdin.readline().split()
    if i[0] == "push":
        queue.append(int(i[1]))
    elif i[0] == "pop":
        if len(queue) > 0:
            print(queue.popleft())
        else:
            print(-1)
    elif i[0] == "size":
        print(len(queue))
    elif i[0] == "empty":
        if len(queue) == 0: 
            print(1)
        else:
            print(0)
    elif i[0] == "front":
        if len(queue) > 0:
            print(queue[0])
        else:
            print(-1)
    elif i[0] == "back":
        if len(queue) > 0:
            print(queue[-1])
        else:
            print(-1)